code/_onclick/hud/parallax.dm

```diff
@@ -248,18 +248,23 @@
 /obj/screen/parallax_layer/proc/update_o(view)
 	if(!view)
 		view = world.view
-	var/list/new_overlays = list()
-	var/count = CEILING(view/(480/world.icon_size), 1)+1
-	for(var/x in -count to count)
-		for(var/y in -count to count)
+
+	var/static/parallax_scaler = world.icon_size / 480
+
+	// Turn the view size into a grid of correctly scaled overlays
+	var/list/viewscales = getviewsize(view)
+	var/countx = CEILING((viewscales[1] / 2) * parallax_scaler, 1) + 1
+	var/county = CEILING((viewscales[2] / 2) * parallax_scaler, 1) + 1
+	var/list/new_overlays = new
+	for(var/x in -countx to countx)
+		for(var/y in -county to county)
 			if(x == 0 && y == 0)
 				continue
 			var/mutable_appearance/texture_overlay = mutable_appearance(icon, icon_state)
-			texture_overlay.transform = matrix(1, 0, x*480, 0, 1, y*480)
+			texture_overlay.transform = matrix(1, 0, x * 480, 0, 1, y * 480)
 			new_overlays += texture_overlay
-
-	overlays = new_overlays
-	view_sized = view
```

Comment:
![image](https://github.com/ParadiseSS13/Paradise/assets/15887760/bf12a2b4-9f46-4b22-9d10-e94e4d32ce30)
This is required else you keep calling the `update_o` proc in `update_parallax`
![image](https://github.com/ParadiseSS13/Paradise/assets/15887760/374095cc-c784-4c3f-b2f2-97d80b1d9a2e)

This solves half of the performance cost of parallax, still 300 CPU is quite high but I'm not sure how much it was before the TM.